app/models/past_link.rb

```diff
@@ -14,7 +15,48 @@ def self.log_link(link)
         })
   end
 
+
+  def set_reaction (response_type, response_text)
+    # Make notification for setter
+    notification_text = "#{link.user.username} loved your post!" if response_type == 'horny'
+    notification_text = "#{link.user.username} did not like your post." if response_type == 'disgust'
+    notification_text = "#{link.user.username} came to your post!" if response_type == 'came'
+    notification_text = "#{notification_text} \"#{response_text}\"" unless response_type.nil?
+    Notification.create user_id: set_by_id, notification_type: :post_response, text: notification_text, link: "/links/#{link.id}"
+
+    # Log reaction in chat sidebar
+    setter_name = set_by&.username || "anon"
+    comment_text = "> loved #{setter_name}'s wallpaper! #{ post_url }" if response_type == 'horny'
+    comment_text = "> hated it. #{ post_url }" if response_type == 'disgust'
+    comment_text = "> came to #{setter_name}'s wallpaper! #{ post_url }" if response_type == 'came'
+    Comment.create user_id: link.user.id, link_id: link.id, content: comment_text
+    Comment.create user_id: link.user.id, link_id: link.id, content: response_text unless response_type.nil?
+
+    # If a came reaction, log an orgasm
+    Nuttracker::Orgasm.create rating: 3, is_ruined: false, user: link.user, caused_by: set_by if response_type == 'came'
+
+    # If a disgust reaction, revert to old wallpaper
+    is_current_post = link.current_past_link == self
+    if response_type == 'disgust'
+      previous_past_post = link.past_links.where.not(id: id).last
+      if is_current_post
+        link.post_url = previous_past_post ? previous_past_post.post_url : nil
+        link.post_thumbnail_url = previous_past_post ? previous_past_post.post_thumbnail_url : nil
+        link.set_by = previous_past_post&.set_by
+      end
+      self.destroy
+      link.save! # failure should only happen here if we've made a mistake, so throw rather than return false
+      return self.destroyed?
+    end
+    self.response_text = response_text
+    self.response_type = response_type
+    return self.save
+  end
+
   after_commit do
+    if response_text_previously_changed? || response_type_previously_changed?
+      broadcast_replace_to "link_response_#{link_id}", target: "link_response_#{link_id}", partial: 'links/reaction', locals: { link: self.link } if self.link.current_past_link == self
```

Comment:
thiiiis might actually be less robust than the previous logic which broadcast updates to all subscribers of the link itself. namely, you can't display the same link reaction on the same page more than once and have them all update.

this never actually happens on the site and i have no idea if you care about this